File: cakk-api/src/main/java/com/cakk/api/dto/request/cake/CakeCreateRequest.java
package com.cakk.api.dto.request.cake;

import java.util.List;

import jakarta.validation.constraints.NotBlank;

import com.cakk.api.mapper.CakeDesignCategoryMapper;
import com.cakk.api.mapper.CakeMapper;
import com.cakk.common.enums.CakeDesignCategory;
import com.cakk.domain.mysql.dto.param.cake.CakeCreateParam;
import com.cakk.domain.mysql.entity.user.User;

public record CakeCreateRequest(
	@NotBlank
	String cakeImageUrl,
	List<CakeDesignCategory> cakeDesignCategories,
	List<String> tagNames
) {

	public CakeCreateParam toParam(User user, Long cakeShopId) {
		return new CakeCreateParam(
			CakeMapper.supplyCakeBy(cakeImageUrl),
			CakeDesignCategoryMapper.supplyCakeCategoryListBy(cakeDesignCategories),
			tagNames,
			user,
			cakeShopId
		);
	}
}

File: cakk-api/src/main/kotlin/com/cakk/api/dto/request/cake/CakeCreateRequest.kt
package com.cakk.api.dto.request.cake

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.Size

import com.cakk.common.enums.CakeDesignCategory

data class CakeCreateRequest(
	@field:NotBlank @field:Size(max = 200)
    val cakeImageUrl: String?,
	@field:NotEmpty
    val cakeDesignCategories: List<CakeDesignCategory>?,
	@field:NotEmpty
    val tagNames: List<String>?
)
